### PR TITLE
refactor: Refactor Core CKF finalization

### PR DESCRIPTION
**Pull request by andiwand**
- Drop `filtered` flag since there are no steps afterwards
- Unify finalization of branches and finding
- Use `end of world` aborter instead `navigation break` flag

### CODE REVIEW

**Comment by noemina** [comment]
Changes look good to me. Thinking about the `end of world` aborter, does this refer to the full Tracking geometry volume, or the user can chose the particular `end of world` to use?

What about the CI failures? I have see a few `TrackFinding   WARNING   Navigation break issued - stopping. This is likely a navigation error`. Is this understood?

File: Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp [hunk 1]
```diff
@@ -394,10 +371,42 @@ class CombinatorialKalmanFilter {
...
-      // Reset propagation state:
-      // - When navigation breaks and there is still active tip present after
-      // recording&removing track tips on current surface
-      if (navigator.navigationBreak(state.navigation) && !result.filtered) {
+      bool isEndOfWorldReached =
+          endOfWorldReached(state, stepper, navigator, logger());
+      bool isPathLimitReached =
+          result.pathLimitReached(state, stepper, navigator, logger());
+      bool isTargetReached = targetReached(state, stepper, navigator, logger());
+      bool isNavigationBreak = navigator.navigationBreak(state.navigation);
```
Comment (AJPfleger) [inline Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp:379]:
```suggestion
      const bool isEndOfWorldReached =
          endOfWorldReached(state, stepper, navigator, logger());
      const bool isPathLimitReached =
          result.pathLimitReached(state, stepper, navigator, logger());
      const bool isTargetReached = targetReached(state, stepper, navigator, logger());
      const bool isNavigationBreak = navigator.navigationBreak(state.navigation);
```
I was double checking, what we are doing with those values. Maybe `const` makes it clearer, that we only need them for the next large scope?

**Review by AJPfleger — APPROVED** [review]

**Comment by andiwand** [comment]
> Changes look good to me. Thinking about the `end of world` aborter, does this refer to the full Tracking geometry volume, or the user can chose the particular `end of world` to use?

Right - this refers to the full geometry at the moment which is not optimal. We do not really have a way right now to confine the propagation to a specific region. There are some ideas and we should definitely start working on this soon.

> What about the CI failures? I have see a few `TrackFinding WARNING Navigation break issued - stopping. This is likely a navigation error`. Is this understood?

I didn't expect this to fire. I will have a look

File: Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp (hunk 1 above)
Comment (andiwand) [reply Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp:379]: sure

**Review by AJPfleger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [2e0ebfed194b7ec337d608707c075c3fc8b946bf]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39145910) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7421994)!

**Pull request merged**